holopy/scattering/theory/lens.py

```diff
@@ -0,0 +1,192 @@
+            prefactor *= ne.evaluate(cls.numexpr_integrand_prefactor3)
+        else:
+            prefactor = np.exp(1j * krho_p * sinth * np.cos(phi - phi_p))
+            prefactor *= np.exp(1j * kz_p * (1 - costh))
+            prefactor *= np.sqrt(costh) * sinth * dphi * dth
+        prefactor *= .5 / np.pi
+        return prefactor
+
+    def _calc_scattering_matrix(self, scatterer, medium_wavevec, medium_index):
+        theta, phi = np.meshgrid(self._theta_pts, self._phi_pts)
+        pts = detector_points(theta=theta.ravel(), phi=phi.ravel())
+        illum_wavelen = 2 * np.pi * medium_index / medium_wavevec
+
+        pts = update_metadata(pts, medium_index=medium_index,
+                              illum_wavelen=illum_wavelen)
+        S = self.theory.calculate_scattering_matrix(scatterer, pts)
+        S = np.conj(S.values.reshape(self.quad_npts_theta,
+                                     self.quad_npts_phi, 2, 2))
+        S = np.swapaxes(S, 0, 1)
+        S1 = S[:, :, 1, 1].reshape(self.quad_npts_theta, self.quad_npts_phi, 1)
+        S2 = S[:, :, 0, 0].reshape(self.quad_npts_theta, self.quad_npts_phi, 1)
+        S3 = S[:, :, 0, 1].reshape(self.quad_npts_theta, self.quad_npts_phi, 1)
+        S4 = S[:, :, 1, 0].reshape(self.quad_npts_theta, self.quad_npts_phi, 1)
+        return S1, S2, S3, S4
+
+    @classmethod
+    def _integrand_prll(cls, prefactor, cosphi, sinphi, S1, S2, S3, S4):
+        if NUMEXPR_INSTALLED:
+            integrand_l = ne.evaluate(cls.numexpr_integrandl)
+        else:
+            integrand_l = prefactor * (cosphi * (cosphi * S2 + sinphi * S3)
+                                     + sinphi * (cosphi * S4 + sinphi * S1))
+        return integrand_l
+
+    @classmethod
+    def _integrand_perp(cls, prefactor, cosphi, sinphi, S1, S2, S3, S4):
+        if NUMEXPR_INSTALLED:
+            integrand_r = ne.evaluate(cls.numexpr_integrandr)
+        else:
+            integrand_r = prefactor * (sinphi * (cosphi * S2 + sinphi * S3)
+                                     - cosphi * (cosphi * S4 + sinphi * S1))
+        return integrand_r
+
+    def _transform_integral_from_lr_to_xyz(self, prll_component, perp_component,
+                                           illum_polarization):
+        pol_angle = np.arctan2(illum_polarization.values[1],
+                               illum_polarization.values[0])
+        parallel = np.array([np.cos(pol_angle), np.sin(pol_angle)])
+        perpendicular = np.array([-np.sin(pol_angle), np.cos(pol_angle)])
+        xyz = np.zeros([3, prll_component.size], dtype='complex')
+        for i in range(2):
+            xyz[i, :] += prll_component * parallel[i]
+            xyz[i, :] += perp_component * perpendicular[i]
+        return xyz
+
+    def _compute_field_prefactor(self, scatterer, medium_wavevec):
+        return -1. * np.exp(1j * medium_wavevec * scatterer.center[2])
+
+    def _raw_scat_matrs(self, *args, **kwargs):
```

Comment:
I don't like that it passes raw scattering matrices through without modification by the lens. I would prefer raising an error that tells you not to use a lens if you are interested in scattering matrices.

Reply:
:+1:

Reply:
I'll just remove this and the Theory will automatically inherent this method from parent which the Lens(Theory) was instantiated with.